internal/cmd/controller/helmops/reconciler/fleethelm_controller.go

```diff
@@ -0,0 +1,413 @@
+			if helmapp.Spec.HelmSecretName != "" {
+				req := types.NamespacedName{Namespace: helmapp.Namespace, Name: helmapp.Spec.HelmSecretName}
+				var err error
+				auth, err = bundlereader.ReadHelmAuthFromSecret(ctx, r.Client, req)
+				if err != nil {
+					return err
+				}
+			}
+			auth.InsecureSkipVerify = helmapp.Spec.InsecureSkipTLSverify
+
+			_, version, err := bundlereader.ChartURLVersion(*bundle.Spec.Helm, auth)
+			if err != nil {
+				return err
+			}
+			bundle.Spec.Helm.Version = version
+		} else {
+			bundle.Spec.Helm.Version = helmapp.Status.Version
+		}
+	} else {
+		bundle.Spec.Helm.Version = helmapp.Spec.Helm.Version
+	}
+
+	return nil
+}
+
+// updateStatusHelm updates the status for the HelmApp resource. It retries on
+// conflict. If the status was updated successfully, it also collects (as in
+// updates) metrics for the resource GitRepo resource.
+func updateStatusHelm(ctx context.Context, c client.Client, req types.NamespacedName, status fleet.HelmAppStatus) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		t := &fleet.HelmApp{}
+		err := c.Get(ctx, req, t)
+		if err != nil {
+			return err
+		}
+
+		// selectively update the status fields this reconciler is responsible for
+		t.Status.Version = status.Version
+
+		// only keep the Ready condition from live status, it's calculated by the status reconciler
+		conds := []genericcondition.GenericCondition{}
+		for _, c := range t.Status.Conditions {
+			if c.Type == "Ready" {
+				conds = append(conds, c)
+				break
+			}
+		}
+		for _, c := range status.Conditions {
+			if c.Type == "Ready" {
+				continue
+			}
+			conds = append(conds, c)
+		}
+		t.Status.Conditions = conds
+
+		err = c.Status().Update(ctx, t)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+}
+
+// updateErrorStatusHelm sets the condition in the status and tries to update the resource
+func updateErrorStatusHelm(ctx context.Context, c client.Client, req types.NamespacedName, status fleet.HelmAppStatus, orgErr error) error {
+	setAcceptedConditionHelm(&status, orgErr)
+	if statusErr := updateStatusHelm(ctx, c, req, status); statusErr != nil {
+		merr := []error{orgErr, fmt.Errorf("failed to update the status: %w", statusErr)}
+		return errutil.NewAggregate(merr)
+	}
+	return orgErr
+}
+
+// setAcceptedCondition sets the condition and updates the timestamp, if the condition changed
+func setAcceptedConditionHelm(status *fleet.HelmAppStatus, err error) {
+	cond := condition.Cond(fleet.HelmAppAcceptedCondition)
+	origStatus := status.DeepCopy()
+	cond.SetError(status, "", fleetutil.IgnoreConflict(err))
+	if !equality.Semantic.DeepEqual(origStatus, status) {
+		cond.LastUpdated(status, time.Now().UTC().Format(time.RFC3339))
+	}
+}
+
+func helmChartSpecChanged(o *fleet.HelmOptions, n *fleet.HelmOptions, statusVersion string) bool {
+	if o == nil {
+		// still not set
+		return true
+	}
+	if o.Repo != n.Repo {
+		// check that the difference is not the / at the end
```

Comment:
They did at some point, because the `ChartURL` function got a pointer parameter and it added the slash.
I changed the function to get a copy instead and this check is no longer needed.

Deleted